src/components/codeContext.tsx

```diff
@@ -277,10 +284,24 @@ export async function createOrgAuthToken({
   }
 }
 
+const getLocallyStoredSelections = (): SelectedCodeTabs => {
+  if (isLocalStorageAvailable()) {
+    return Object.fromEntries(
+      Object.entries(localStorage).filter(([key]) => key.startsWith('Tabgroup:'))
+    );
+  }
+  return {};
+};
+
 export function CodeContextProvider({children}: {children: React.ReactNode}) {
   const [codeKeywords, setCodeKeywords] = useState(cachedCodeKeywords ?? DEFAULTS);
-
   const [isLoading, setIsLoading] = useState<boolean>(cachedCodeKeywords ? false : true);
+  const [storedCodeSelection, setStoredCodeSelection] = useState<SelectedCodeTabs>({});
+
+  // populate state using localstorage
+  useEffect(() => {
+    setStoredCodeSelection(getLocallyStoredSelections());
+  }, []);
```

Comment:
with this I think we can get rid of the isLocalStorageAvailable check right?

Reply:
I would just leave it tbh

Reply:
sounds good